src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs

```diff
@@ -308,7 +316,8 @@ public GraphQLExecutor NewExecutor(IRequestDeserializer requestDeserializer = nu
                     { typeof(IDependencyInjector).FullName ?? nameof(IDependencyInjector), dependencyInjector },
                 },
                 ValidationRules = validationRules.Any() ? validationRules : null,
-                ComplexityConfiguration = complexityConfiguration,
+                //TODO: how to pass through complexityConfiguration
+                // ComplexityConfiguration = complexityConfiguration,
```

Comment:
@Shane32 how do i convert this over?

Reply:
Use `ComplexityValidationRule` that accepts an instance of `ComplexityConfiguration`.

Reply:
Depends if the configuration assumes the use of `AddComplexityAnalyzer()`; if so there should be no need to set this here.  I'm pretty sure that's not the case with this library.

So, if `IComplexityAnalyzer` is assumed to be registered in DI, then maybe something of this nature:

```cs
var analyzer = provider.GetService<IComplexityAnalyzer>();
if (analyzer != null)
{
    validationRules = validationRules.Append(new ComplexityValidationRule(complexityConfiguration, analyzer));
}
```

Or perhaps something like this:

```cs
var analyzer = provider.GetService<IComplexityAnalyzer>();
if (analyzer != null && !validationRules.Any(rule => rule is ComplexityValidationRule))
{
    validationRules = validationRules.Append(new ComplexityValidationRule(complexityConfiguration, analyzer));
}
```

Or if the service provider is not available, perhaps this: (which is not deprecated, unlike the preceding options)

```cs
validationRules = validationRules.Append(new ComplexityValidationRule(complexityConfiguration));
```

Perhaps there should be some check of the complexity configuration to be sure that adding the rule is warranted.

Reply:
Probably also need something like this prior to that code for 100% compatibility:

```cs
validationRules = validationRules.Any() ? validationRules : GraphQL.Validation.DocumentValidator.CoreRules;
```

Reply:
there is no `provider` in this scope

Reply:
i dont understand how ` enableValidation`, `complexityConfiguration`, `IComplexityAnalyzer`, and `DocumentValidator.CoreRules` should interact. can one of you provide a the code to what `ValidationRules` should be passed to `ExecutionOptions`

Reply:
I took a look.  Turns out that `complexityConfiguration` is `null` unless someone calls `WithComplexityConfiguration`.  So we just check if the configuration is not null, and then append the validation rule.  I'll push a fix here.

Reply:
Fixed.  Right or wrong, it now behaves the same as it did in GraphQL.NET v5.